File: app/src/main/java/com/dahyeon/mirim/activities/LoginActivity.java
package com.dahyeon.mirim.activities;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.CheckBox;
import android.widget.EditText;
import android.widget.TextView;
import com.dahyeon.mirim.R;

public class LoginActivity extends AppCompatActivity {

    //<로그인 페이지>
    EditText email_in; //이메일 입력
    EditText password_in; //패스워드 입력

    CheckBox ck_in; //자동로그인 체크박스
    Button btn_in; //로그인 버튼
    TextView text_up; //'아직 회원이 아니신가요?' 텍스트뷰
    //TextView forgot_passwd; //비밀번호 찾기 텍스트뷰 => 페이지 아직 안만들어져있음

    TextView be_in; //로그인 안되는 이유 나타내는 텍스트뷰

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        ck_in = findViewById(R.id.ck_auto);
        btn_in = findViewById(R.id.btn_in);
        text_up = findViewById(R.id.text_in);

        be_in = findViewById(R.id.because_in);

        btn_in.setOnClickListener(new View.OnClickListener() { //로그인 버튼 - DB에서 맞는지 체크 필요
            @Override
            public void onClick(View v) {
                //DB에서 체크
                //DB에 없는 이메일, 비밀번호라면 be_in.setText("존재하지 않는 이메일, 비밀번호입니다");
                Intent intent = new Intent(getApplicationContext(), MainActivity.class);
                startActivity(intent);
                overridePendingTransition(android.R.anim.slide_in_left, android.R.anim.fade_in);
            }
        });

        text_up.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) { //회원가입 해주세요
                Intent intent = new Intent(getApplicationContext(), JoinActivity.class);
                startActivity(intent);
                overridePendingTransition(android.R.anim.slide_in_left, android.R.anim.fade_in);
            }
        });

        btn_in.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                email_in = findViewById(R.id.email_in);
                password_in = findViewById(R.id.password_in);

                Intent intent = new Intent(getApplicationContext(), MainActivity.class);
                startActivity(intent);
            }
        });
    }
}

File: app/src/main/java/com/dahyeon/mirim/activities/LoginActivity.kt
package com.dahyeon.mirim.activities

import android.content.Intent
import android.os.Bundle
import android.util.Log
import com.dahyeon.mirim.R
import com.dahyeon.mirim.models.AuthModel
import com.dahyeon.mirim.network.Connector
import com.dahyeon.mirim.network.Res
import com.dahyeon.mirim.util.BaseActivity
import kotlinx.android.synthetic.main.activity_login.*

class LoginActivity : BaseActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        button_main_login.setOnClickListener {
            val emailPrefix = edittext_main_email.text
            val password = edittext_main_password.text

            if (emailPrefix.isEmpty() || password.isEmpty()) {
                showToast("값을 모두 입력하세요.")
            } else {
                Log.d("@@@@@", "before auth")
                Log.d("@@@@@", hashMapOf(
                    "email" to "$emailPrefix@e-mirim.hs.kr",
                    "pw" to password
                ).toString())

                Connector.api.auth(
                    hashMapOf(
                        "email" to "$emailPrefix@e-mirim.hs.kr",
                        "pw" to password
                    )
                ).enqueue(object: Res<AuthModel>(this) {
                    override fun callBack(code: Int, body: AuthModel?) {
                        Log.d("@@@@@", "callback")
                        when(code) {
                            200 -> {
                                saveToken(body!!.token)
                                saveToken(body.refreshToken!!, false)
                                null
                            }
                            401 -> {
                                "이메일 또는 비밀번호를 확인하세요."
                            }
                            else -> {
                                "오류: $code"
                            }
                        }.let {
                            if(it != null) {
                                showToast(it)
                            } else {
                                startActivity(Intent(applicationContext, MainActivity::class.java))
                            }
                        }
                    }
                })
            }
        }
    }
}
